ext/grid.jl

```diff
@@ -0,0 +1,158 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+function Makie.plot!(plot::Viz{<:Tuple{Grid}})
+  grid = plot[:object]
+  color = plot[:color]
+  alpha = plot[:alpha]
+  colorscheme = plot[:colorscheme]
+  
+  # process color spec into colorant
+  colorant = Makie.@lift process($color, $colorscheme, $alpha)
+  
+  # number of vertices and colors
+  nv = Makie.@lift nvertices($grid)
+  nc = Makie.@lift $colorant isa AbstractVector ? length($colorant) : 1
+  
+  T = typeof(grid[])
+  Dim = embeddim(grid[])
+  if Dim == 1
+    vizgrid1D!(T, plot, colorant, nv, nc)
+  elseif Dim == 2
+    vizgrid2D!(T, plot, colorant, nv, nc)
+  elseif Dim == 3
+    vizgrid3D!(T, plot, colorant, nv, nc)
+  end
+end
+
+vizgrid1D!(::Type{<:Grid}, plot, colorant, nv, nc) = vizmesh1D!(plot)
+
+vizgrid2D!(::Type{<:Grid}, plot, colorant, nv, nc) = vizmesh2D!(plot)
+
+function vizgrid2D!(::Type{<:Union{CartesianGrid,RectilinearGrid}}, plot, colorant, nv, nc)
```

Comment:
Can we refactor all the vizgrid methods to accept only the grid type and the plot? I think it would be more useful to keep the colorant and nv and nc inside as an implementation detail. That way we can reuse these methods in other places if necessary.